src/NuGet.Core/NuGet.Versioning/VersionRangeFactory.cs

```diff
@@ -376,6 +376,146 @@ public static VersionRange Combine(IEnumerable<VersionRange> ranges, IVersionCom
         }
 
         /// <summary>
+        /// Returns the greatest common range that satisfies all given ranges.
+        /// </summary>
+        public static VersionRange CommonSubSet(IEnumerable<VersionRange> ranges)
+        {
+            return CommonSubSet(ranges, VersionComparer.Default);
+        }
+
+        /// <summary>
+        /// Returns the greatest common range that satisfies all given ranges.
+        /// </summary>
+        public static VersionRange CommonSubSet(IEnumerable<VersionRange> ranges, IVersionComparer comparer)
+        {
+            if (ranges == null)
+            {
+                throw new ArgumentNullException(nameof(ranges));
+            }
+
+            if (comparer == null)
+            {
+                throw new ArgumentNullException(nameof(comparer));
+            }
+
+            // Default to None for empty lists
+            var result = None;
+
+            // Remove zero width ranges. Ex: (1.0.0, 1.0.0)
+            // This includes VersionRange.None and any other ranges that satisfy zero versions
+            ranges = ranges.Where(range => HasValidRange(range));
```

Comment:
Invalid ranges should cause this method to return None instead of being ignored. Add a test for this also